packages/compiler/core/program.ts

```diff
@@ -621,6 +649,55 @@ export async function compile(
     }
   }
 
+  function computeLibraryMetadata(module: ModuleResolutionResult): LibraryMetadata {
+    if (module.type === "file") {
+      return {};
+    }
+
+    const metadata: LibraryMetadata = {
+      name: module.manifest.name,
+    };
+
+    if (module.manifest.homepage) {
+      metadata.homepage = module.manifest.homepage;
+    }
+    if (module.manifest.bugs?.url) {
+      metadata.bugs = { url: module.manifest.bugs?.url };
+    }
+
+    return metadata;
+  }
+
+  /**
+   * @param emitter Emitter ref to run
+   */
+  async function runEmitter(emitter: EmitterRef) {
```

Comment:
Would it make sense to have a similar behavior for everything(decorators, onValidate)?

Reply:
I think we should be consistent, yes. I don't think there's anything special about onEmit throwing vs. a decorator or onValidate with respect to an exception being raised.

But I would prefer if these stayed scary fail-fast bugs and not diagnostics. I don't want to have to think about what happens next after one of these fails, and it's worse for decorators than emit/onvalidate. 

It's also worth keeping in mind that the exception could also really be a bug in the core. We call decorator, decorator/onemit/onvalidate, which calls compiler API, which hits a bug and throws. And in that case, I especially don't want to be chugging along calling more things, hitting the same bug, and blaming more emitters, etc.

Reply:
Yeah that true, à crash like that shouldn’t continue. Maybe instead I can throw a known exception which has the right info to log

Reply:
Filed https://github.com/microsoft/cadl/issues/1120